File: 백준/Silver/9465. 스티커/스티커.cs
using System;
using System.Linq;

class Program
{
    static void Main() {
        int t = int.Parse(Console.ReadLine());
        while((t--)!=0)
        {
            int[,] dp = new int[2,100001];
            int n = int.Parse(Console.ReadLine());
            string[] s = Console.ReadLine().Split(' ');
            for(int i=0;i<n;i++) dp[0,i] = int.Parse(s[i]);
            s = Console.ReadLine().Split(' ');
            for(int i=0;i<n;i++) dp[1,i] = int.Parse(s[i]);
            dp[0,1] += dp[1,0];
            dp[1,1] += dp[0,0];
            
            for(int i=2;i<n;i++)
            {
                dp[0,i] += Math.Max(dp[1,i-1],dp[1,i-2]);
                dp[1,i] += Math.Max(dp[0,i-1],dp[0,i-2]);
            }
            Console.WriteLine(Math.Max(dp[0,n-1],dp[1,n-1]));
        }
    }
}

File: 백준/Silver/9465. 스티커/스티커.py
t = int(input())

for i in range(t):
  n = int(input())
  d = [list(map(int,input().split())) for i in range(2)]
  if n!=1:
      d[0][1] += d[1][0]
      d[1][1] += d[0][0]
  
  for j in range(2, n):
    d[0][j] += max(d[1][j - 1], d[1][j - 2])
    d[1][j] += max(d[0][j - 1], d[0][j - 2])
    
  print(max(d[0][n - 1], d[1][n - 1]))